src/unix/darwin.c

```diff
@@ -65,28 +65,33 @@ uint64_t uv__hrtime(uv_clocktype_t type) {
 
 
 int uv_exepath(char* buffer, size_t* size) {
-  uint32_t usize;
-  int result;
-  char* path;
-  char* fullpath;
+  /* realpath(exepath) may be > PATH_MAX so double it to be on the safe side. */
+  char abspath[PATH_MAX * 2 + 1];
```

Comment:
Wouldn't the maximum size for realpath be `PATH_MAX` plus the maximum length of a filename (on HFS+ I think that's 255)?